src/System.Net.Security/src/System/Net/Security/NegotiateStream.cs

```diff
@@ -502,6 +502,11 @@ public override void Flush()
 #endif
         }
 
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return InnerStream.FlushAsync(cancellationToken);
```

Comment:
I've not used that, but my guess is that it was never intended to be used across threads.